receiver/googlecloudspannerreceiver/config.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlecloudspannerreceiver
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+
+	TopMetricsQueryMaxRows int       `mapstructure:"top_metrics_query_max_rows"`
+	Projects               []Project `mapstructure:"projects"`
+}
+
+type Project struct {
+	ID                string     `mapstructure:"project_id"`
+	ServiceAccountKey string     `mapstructure:"service_account_key"`
+	Instances         []Instance `mapstructure:"instances"`
+}
+
+type Instance struct {
+	ID        string   `mapstructure:"instance_id"`
+	Databases []string `mapstructure:"databases"`
+}
+
+func (config *Config) Validate() error {
+	if config.CollectionInterval.Seconds() < 60 {
+		return fmt.Errorf("%v `collection_interval` must be not lower than 60s, current value is %vs",
+			config.ID(), config.CollectionInterval.Seconds())
+	}
+
+	if config.TopMetricsQueryMaxRows <= 0 {
+		return fmt.Errorf("%v `top_metrics_query_max_rows` must be positive: %v", config.ID(), config.TopMetricsQueryMaxRows)
+	}
+
+	if config.TopMetricsQueryMaxRows > 100 {
+		return fmt.Errorf("%v `top_metrics_query_max_rows` must be not greater than 100, current value is %v",
+			config.ID(), config.TopMetricsQueryMaxRows)
+	}
+
+	if len(config.Projects) <= 0 {
+		return fmt.Errorf("%v missing required field %v or its value is empty", config.ID(), "`projects`")
+	}
+
+	for _, project := range config.Projects {
+		if err := project.Validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (project Project) Validate() error {
```

Comment:
Make the sub-validation helpers private instead of public (`validate`)

Reply:
Ok, will do

Reply:
Done.